GetUpChuckUnityProject/Assets/Scripts/Game World/Plate.cs

```diff
@@ -0,0 +1,24 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Plate : Obstacle, MonoBehaviour {
+
+	public int weightCheck;
+	bool pressed = false;
+
+	void OnCollisionEnter2D(Collision2D coll) {
+		if (coll.gameObject.tag == "Player") { condCheck(); }
```

Comment:
use 
coll.gameObject.tag.Equals("Key")
for string comparison

condCheck needs to be passed an inventory